마이페이지 화면구성을 진행합니다.

jaychang99 opened the pull request:
작성자: @jaychang99

Related to 

## 체크 리스트

- [x] 적절한 제목으로 수정했나요?
- [] 상단에 이슈 번호를 기입했나요?
- [x] Target Branch를 올바르게 설정했나요?
- [x] Label을 알맞게 설정했나요?

## 작업 내역

- 마이페이지 메인과 마이페이지에서 랜딩될 수 있는 모든 페이지 (수정, 삭제 확인, 삭제 등) 를 화면구성합니다. 
- 아래 각 이미지를 참고하시기 바랍니다. 

### 작업 중 공통 컴포넌트 변경 내역
- 다른 곳들의 영역 중 Props 를 받지 않는 데에도 interface 가 정의된 곳의 삭제를 진행하였습니다. 
- `<TextInput />` 컴포넌트가 이제 `values` props 의 변화도 감지하도록 하여 초기값을 가져오는 비동기 통신에 의한 데이터 변화를 감지할 수 있도록 하였습니다. 
- `<Button />` 컴포넌트가 외부의 `className` 도 받을 수 있도록 수정하였습니다. 
- 날짜 연산을 위한 `dayjs` 날짜 관리 라이브러리를 설치하여 보았습니다. 


![image](https://github.com/SystemConsultantGroup/foodhub-frontend/assets/86560973/095f45c0-d39c-41e5-b886-4cfc1247a498)

![image](https://github.com/SystemConsultantGroup/foodhub-frontend/assets/86560973/4b7cd814-fd1d-4195-97d4-bbf1d5b40cc4)

![image](https://github.com/SystemConsultantGroup/foodhub-frontend/assets/86560973/7db9e68b-c0e0-4ff8-82a5-907cfdb2549c)

![image](https://github.com/SystemConsultantGroup/foodhub-frontend/assets/86560973/14ead684-8de9-4cad-af94-6736faf586ce)


## 비고

- `DatePicker` 가 아직 개발되지 않았는데 출생년도 선택이니까 그냥 드롭다운으로 해도 되지 않을까 라는 생각이 들었습니다. 그렇게 바꾸어 볼까요?
- validation 은 아직 명확한 정책이 없어 [탈퇴] 문구를 제외하고는 적용하지 않은 상태입니다.

jaychang99 committed:
feat(Mypage): 마이페이지 페이지 골격 구성

jaychang99 committed:
Merge branch 'feature/page-structure-setup' into feature/page-mypage

jaychang99 committed:
feat: ProfileImage 에 사이즈 추가할 수 있도록 변경

jaychang99 committed:
feat(MyPage): 마이페이지 상단 프로필까지 퍼블리싱

jaychang99 committed:
feat(Mypage): 마이페이지 내 정보 및 상세 영역, 단체 정보 제외 퍼블리싱 완료

jaychang99 committed:
feat(Mypage): 마이페이지 스타일 변경

jaychang99 committed:
sync: Merge branch 'develop'

Conflicts:
	src/components/profileImage/ProfileImage.tsx

jaychang99 committed:
fix(Mypage): 마이페이지 내 정보 수정 버튼 공통 컴포넌트로 대체

jaychang99 committed:
fix(Button): 버튼이 외부 className 도 받을 수 있도록 수정

jaychang99 committed:
feat(MyPage): 마이페이지 수정 페이지 기본 레이아웃 구현

jaychang99 committed:
fix(TextInput): 비동기 api 호출에 의한 value 변화를 감지하지 못하는 문제 수정

jaychang99 committed:
feat(MyPage): 마이페이지 정보 수정 페이지 생년월일 제외 레이아웃 및 모킹 구성 완료

jaychang99 committed:
feat(MyPage): 삭제 확인 페이지 생성 완료

jaychang99 committed:
refactor(Mypage): 세부기능별 폴더구조 개편

jaychang99 committed:
refactor: Props 받지 않는데 정의된 곳 제거

jaychang99 committed:
feat(Mypage): 회원탈퇴 완료 페이지 생성

jaychang99 committed:
feat(Mypage): 마이페이지의 '내 정보 수정' 버튼을 primary 버튼에서 text 버튼으로 격하'

jaychang99 committed:
feat(Mypage): 마이페이지 메인에 가입된 단체 없을 때 CTA 영역 추가

jaychang99 committed:
feat(Mypage): 마이페이지 메인에서도 initial mockup data 로 정보 불러올 수 있도록 세팅

sera2222 commented:
출생 연도만 선택하는 거면 dropdown으로 구현을 하는게 좋을 것 같습니다

sera2222 commented:
마이페이지에 로그아웃 버튼을 하나 만들어야 할 것 같습니다!

jaychang99 commented:
@sera2002 오 넵 세라님 드롭다운으로 출생년도 구현 및 로그아웃 버튼 추가하겠습니다! 좋은 의견 감사합니당

jaychang99 commented:
@sera2002  세라님 그러고보니 로그아웃 버튼은 상단 내비게이션 바 어디엔가 추가될 것 같아서 마이페이지에 없어도 될 것 같은데 어떻게 생각하시나용?

jaychang99 committed:
feat(Dropdown): 드롭다운의 최대 높이를 설정하여, 이를 넘어갈 시 스크롤되도록 할 수 있는 옵션 추가

jaychang99 committed:
feat(MyPage): 마이페이지에 출생년도 설정 드롭다운 추가

jaychang99 commented:
> 출생 연도만 선택하는 거면 dropdown으로 구현을 하는게 좋을 것 같습니다

@sera2002 드롭다운으로 구현 완료하였습니다!

![image](https://github.com/SystemConsultantGroup/foodhub-frontend/assets/86560973/5eb65916-c008-4cb5-9c49-541cfb3e2968)

sera2222 commented:
> @sera2002 세라님 그러고보니 로그아웃 버튼은 상단 내비게이션 바 어디엔가 추가될 것 같아서 마이페이지에 없어도 될 것 같은데 어떻게 생각하시나용?

음.. 이 서비스를 이용할 때 로그아웃/재로그인이 많이 일어나진 않을 것 같아서 굳이 상단 내비게이션 바에 로그아웃 버튼을 항상 노출을 시켜놓을 필요는 없을 것 같습니다.

jaychang99 commented:
> > @sera2002 세라님 그러고보니 로그아웃 버튼은 상단 내비게이션 바 어디엔가 추가될 것 같아서 마이페이지에 없어도 될 것 같은데 어떻게 생각하시나용?
> 
> 음.. 이 서비스를 이용할 때 로그아웃/재로그인이 많이 일어나진 않을 것 같아서 굳이 상단 내비게이션 바에 로그아웃 버튼을 항상 노출을 시켜놓을 필요는 없을 것 같습니다.

아하 넵넵! 마이페이지에 추가하겠습니다!

jaychang99 committed:
feat(MyPage): 마이페이지에 로그아웃 버튼 추가

kanghoon-dev commented on src/components/dropdown/Dropdown.tsx:
```diff
@@ -60,7 +67,11 @@ const Dropdown = ({ children, label, value, onSelectValueChange }: Props) => {
-    <EmotionWrapper isOpen={isOpen} isCompleted={selectedOptionValue ? true : false}>
+    <EmotionWrapper
+      isOpen={isOpen}
+      isCompleted={selectedOptionValue ? true : false}
```
요거 !!selectedOptionValue와 같이 줄이는거 어떨까요?

jaychang99 replied:
오 넵 반영하였습니다!

kanghoon-dev replied:
확인이 늦었습니다...확인했습니다!

kanghoon-dev commented on src/feature/mypage/mypage.delete/components/MypageDeleteConfirmForm.tsx:
```diff
@@ -0,0 +1,66 @@
+
+const MypageDeleteConfirmForm = () => {
+  const { push } = useRouter();
+  const deleteSuccessLink = "/mypage/delete/success";
```
이거 전체 대문자로 바꾸고 함수 밖으로 빼두면 어떨까요?

kanghoon-dev replied:
같은 파일 내에서라도 분리해두면 보기 편할거 같아요

jaychang99 replied:
@hoon5083 오 좋은생각인 것 같습니다. 오히려 아예 공통으로 상수같은 데에 관리하면 어떨까요?

kanghoon-dev replied:
네네 저도 그게 좋을거 같아요!
졸논에서 API 엔드포인트 관리했듯 이제는 페이지까지 관리를....

jaychang99 replied:
@hoon5083 앜ㅋㅋㅋ 넵넵 우선 이번 브랜치에 있는 모든 링크는 `src/constant/link.ts` 에서 관리하는 것으로 바꾸어 보았습니다!

jaychang99 replied:
아 그냥 상수로 관리했는데 구조화를 좀 시킬까요? 뭔가 구조화 시키기엔 이거는 API 엔드포인트가 아니라서 모두 규칙이 있는 것은 아니라서 개별 상수로 놔두는 것도 괜찮을 것 같기도 한데 어떻게 생각하시나용?

kanghoon-dev replied:
네네 구조화에 시간 들이는것보다는 일단 분리하는 정도면 충분할 것 같습니다!

kanghoon-dev commented on src/feature/mypage/mypage.main/components/MyPageBasicInfo.tsx:
```diff
@@ -0,0 +1,38 @@
+  const age = dayjs().year() - Number(birthYear);
+  const ageText = `${age}세`;
+
+  const genderText = GENDER_LOOKUP_TABLE[gender] ?? "입력하지 않음";
```
이거 테이블에 있는 U랑 "입력하지 않음"의 차이가 어떻게 될까요?

jaychang99 replied:
@hoon5083 '입력하지 않음' 은 말그대로 입력하지 않은 상태, U 는 명시적으로 어떤 성별인지 드러내고 싶지 않은 상태를 의미한다고 생각하였습니다.

kanghoon-dev commented on src/feature/mypage/mypage.edit/views/ViewMypageEdit.tsx:
```diff
@@ -0,0 +1,122 @@
+            </Option>
+          ))}
+        </Dropdown>
+        {/* TODO => placeholder 가 적용되지 않는 문제 핵결*/}
+        <Dropdown label="성별" placeholder="성별을 선택해주세요!" value={gender.value}>
```
혹시 원인 대략 파악된거 있을까요?

jaychang99 replied:
@hoon5083 오 방금 파악 완료했습니다!

아래 이미지를 봐주세요. 

![image](https://github.com/SystemConsultantGroup/foodhub-frontend/assets/86560973/0f98831b-a904-4287-8e6b-0bd8310875ed)


![image](https://github.com/SystemConsultantGroup/foodhub-frontend/assets/86560973/85fb0d54-287d-4c5b-8061-4fe89c9f91cc)


현재 `<Dropdown />` 이 받을 수 있는 옵션은 `placeholder` 와 `placeHolder` 의 2가지입니다... 😅 

`placeholder` 은 `HTMLAttribute<HTMLDivElement>` 에서 오는 html (react.d.ts) 기본 속성 타입이며 `<Dropdown />` 이 개발될 때  `placeHolder` 도 추가된 것 같습니다. 그래서 여기 코드에서는 `placeholder` 을 써도 이 타입이 있긴 하니까 타입에러는 나지 않았는데, 이부분을 받아 쓰고 있는 곳이 없어서 반영이 안 되었습니다. `placeHolder` 로 반영을 하고 나서 적용이 되었습니다. 아예 드롭다운에서 이 부분을 placeholder 프롭으로 바꾸면 어떨까 싶습니다... 만 그러면 이 이후 발생할 모든 컨플릭이 무섭네요.. cc. @sera2002

kanghoon-dev replied:
엄.... 일단 머지하고 전체적으로 수정하는 PR올리는게 어떨까요?

kanghoon-dev commented on src/feature/mypage/mypage.edit/views/ViewMypageEdit.tsx:
```diff
@@ -0,0 +1,122 @@
+  const handleChangeDescription = useCallback(handleChangeInput("introduction"), []);
+
+  const { nickname, introduction, gender, birthYear } = formValues;
+  const deleteUserAccountLink = "/mypage/delete";
```
이것도 분리해두면 좋을거 같습니다

jaychang99 replied:
처리했씁니다!

kanghoon-dev reviewed (commented):
수고하셨습니다! 수정 제안 사항 몇가지 있습니다

jaychang99 committed:
sync: Merge branch 'develop'

Conflicts:
	src/components/button/Button.tsx
	src/components/dropdown/Dropdown.tsx

jaychang99 committed:
refactor(Dropdown): boolean 화 해주는 코드 간략화

jaychang99 committed:
refactor: 내부적인 링크 이동 url 모두 constants/link.ts 에서 상수로 관리

jaychang99 reviewed (commented):
@hoon5083 리뷰 반영 완료하였습니다 감사합니다!

sera2222 commented on src/feature/mypage/mypage.delete/components/MypageDeleteConfirmForm.tsx:
```diff
@@ -0,0 +1,66 @@
+
+  button {
+    margin-top: 32px;
+    // TODO: danger Button  머지되면 버튼에 danger prop 부여
```
danger props 추가된거 머지된 상태여서 여기 수정하시면 될 것 같습니다!

jaychang99 replied:
@sera2002 오 예리한 포인트군요 변경하였습니다!!

sera2222 reviewed (approved):
수고하셨습니다!

jaychang99 committed:
refactor(Mypage): 탈퇴하기 버튼에 Button 의 danger prop 이용하는 것으로 변경

kanghoon-dev reviewed (approved):
리뷰 사항 반영확인이 많이 늦었습니다 :( 수고하셨습니다!
